internal/commands/app/create.go

```diff
@@ -98,20 +101,39 @@ func (cmd *CommandCreate) Handler(profile *user.Profile, ui terminal.UI, clients
 		return err
 	}
 
-	var dsCluster dataSourceCluster
-	if cmd.inputs.Cluster != "" {
-		dsCluster, err = cmd.inputs.resolveCluster(clients.Atlas, groupID)
+	var nonExistingDataSources []string
+
+	var dsClusters []dataSourceCluster
+	if len(cmd.inputs.Clusters) > 0 {
+		var nonExistingClusters []string
+		dsClusters, nonExistingClusters, err = cmd.inputs.resolveClusters(ui, clients.Atlas, groupID)
 		if err != nil {
 			return err
 		}
+		nonExistingDataSources = append(nonExistingDataSources, nonExistingClusters...)
 	}
 
-	var dsDataLake dataSourceDataLake
-	if cmd.inputs.DataLake != "" {
-		dsDataLake, err = cmd.inputs.resolveDataLake(clients.Atlas, groupID)
+	var dsDatalakes []dataSourceDatalake
+	if len(cmd.inputs.Datalakes) > 0 {
+		var nonExistingDatalakes []string
+		dsDatalakes, nonExistingDatalakes, err = cmd.inputs.resolveDatalakes(ui, clients.Atlas, groupID)
 		if err != nil {
 			return err
 		}
+		nonExistingDataSources = append(nonExistingDataSources, nonExistingDatalakes...)
+	}
+
+	if len(nonExistingDataSources) > 0 {
```

Comment:
after initializing the `nonExistingDataSources` with an initial capacity here, for display purposes I would then loop through each "missing" array:

```go
for _, missingCluster := range dsClustersMissing {
  nonExistingDataSources = append(nonExistingDataSources, fmt.Sprintf("'%s'", missingCluster))
  // likewise for datalakes
}
```

i don't think we allow spaces in cluster or datalake names, but if we did those quotes will go a long way here